features/facebookImport/src/components/activitiesMinistory/activitiesMinistory.jsx

```diff
@@ -0,0 +1,182 @@
+import React, { useState } from "react";
+
+import i18n from "../../i18n";
+
+import BarChartHorizontal from "../dataViz/barChartHorizontal.jsx";
+
+import { fillArray } from "../dataViz/utils.jsx";
+
+import "./activitiesMinistory.css";
+import "./datePicker.css";
+import "../tabs/tabs.css";
+
+const monthsAbbreviation = [
+    "Jan",
+    "Feb",
+    "Mar",
+    "Apr",
+    "Mai",
+    "Jun",
+    "Jul",
+    "Aug",
+    "Sep",
+    "Oct",
+    "Nov",
+    "Dec",
+];
+
+//TODO let this support actual dates not only years
+const Datepicker = ({ year, yearRange, onYearChange }) => {
+    const rangeIndex = yearRange.indexOf(year);
+    return (
+        <div className="datepicker">
+            {rangeIndex != 0 ? (
+                <button
+                    className="arrow left"
+                    onClick={() => onYearChange(yearRange[rangeIndex - 1])}
+                >
+                    <img src="./images/angle-left.svg" alt="arrow-left" />
+                </button>
+            ) : (
+                <div className="arrow filler"></div>
+            )}
+            <div className="year">{year}</div>
+            {rangeIndex != yearRange.length - 1 ? (
+                <button
+                    className="arrow right"
+                    onClick={() => onYearChange(yearRange[rangeIndex + 1])}
+                >
+                    <img src="./images/angle-right.svg" alt="arrow-right" />
+                </button>
+            ) : (
+                <div className="arrow filler"></div>
+            )}
+        </div>
+    );
+};
+
+const ActivitiesMinistory = ({ totalEvents }) => {
+    const yearRange = fillArray(
+        Object.keys(totalEvents).filter((key) => key != "total")
```

Comment:
We could also have two attributes `total`  and `values` in `totalEvents ` object so we do not filter by key.

Reply:
Yeah will do that!

Reply:
The generation of totalEvents now became even worse, but wokring with it makes a lot more sense! 😅